Describe JSON-LD vocabulary used, and add examples of specific use ...

gkellogg opened the pull request:
..  and .nterpretation in common properties.

This does not yet describe normalizing, which is handled in issue #269, or CSVW-specific RDF processing instructions.

Pulling this will fix #203.

gkellogg committed:
Describe JSON-LD vocabulary used, and add examples of specific use and interpretation in common properties.
This does not yet describe normalizing, which is handled in issue #269, or CSVW-specific RDF processing instructions.

gkellogg committed:
Merge remote-tracking branch 'origin/gh-pages' into issue-203-jsonld-restrictions

gkellogg committed:
Create an appendix section on normalizing JSON-LD used within common properties and notes. This fixes #269.

gkellogg committed:
Handle more corner cases in normalizing JSON-LD content to ensure that values conform to our restrictions, validators issue appropriate warnings, and the result may be expanded to RDF.
This fixes #142.

gkellogg committed:
Merge remote-tracking branch 'origin/gh-pages' into issue-203-jsonld-restrictions

Conflicts:
	metadata/index.html

Commit:
rephrasing & rejigging of examples

gkellogg committed:
* In order to ensure interoperability, when normalizing JSON-LD, validators and transformers MUST raise an error if anything outside the scope of our dialect is found when merging. (seems a bit draconian).
* Move JSON-LD Normalization from appendix to a sub-section of Merge.

This relates to issue #203.

Commit:
described restrictions on common property values

Commit:
moved around & simplified normalisation

including integrating all normalisation in one algorithm, moving the
normalisation examples into the normalisation section, introducing a
subsection on the names of common properties, removing repetitious
constraints

Commit:
fixed examples in model document

showing post-normalisation values in the model

JeniT commented:
I've tried to pull things together a bit, particularly around normalisation. I've described the restrictions on the types of values allowed for common properties in the section defining common properties, and pulled the examples about normalisation of the values of common properties into the section on normalisation.

I've tried to make the text less JSON-LD focused. Most implementers won't be implementing on top of a JSON-LD processor, so describing it in terms of JSON-LD adds an unnecessary overhead to understanding.

I am a bit concerned that there is a remaining discrepancy between the description of the restrictions on the values of common properties in the section on common properties and the description in the JSON-LD Dialect appendix. Please double check.

gkellogg committed:
Update Values of Common Properties to indicate differences of possible `@type` values depending on if used with or without `@value`.
Describe restrictions on `@language` only when used with `@value`, as it's not appropriate elsewhere.

gkellogg committed:
Merge remote-tracking branch 'origin/gh-pages' into issue-203-jsonld-restrictions

# Conflicts:
#	metadata/index.html
